src/main/java/com/aws/greengrass/clientdevices/auth/metrics/handlers/MetricsConfigurationChangedHandler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.metrics.handlers;
+
+import com.aws.greengrass.clientdevices.auth.api.DomainEvents;
+import com.aws.greengrass.clientdevices.auth.configuration.events.MetricsConfigurationChanged;
+import com.aws.greengrass.clientdevices.auth.metrics.MetricsEmitter;
+
+import java.util.function.Consumer;
+import javax.inject.Inject;
+
+public class MetricsConfigurationChangedHandler implements Consumer<MetricsConfigurationChanged> {
+    private final DomainEvents domainEvents;
+    private final MetricsEmitter metricsEmitter;
+
+    /**
+     * Constructor for Metrics Configuration Changed Handler.
+     *
+     * @param domainEvents   Domain event router
+     * @param metricsEmitter Metrics emitter
+     */
+    @Inject
+    public MetricsConfigurationChangedHandler(DomainEvents domainEvents, MetricsEmitter metricsEmitter) {
+        this.domainEvents = domainEvents;
+        this.metricsEmitter = metricsEmitter;
+    }
+
+    /**
+     * Listen for changes to the metrics configuration.
+     */
+    public void listen() {
+        domainEvents.registerListener(this, MetricsConfigurationChanged.class);
+    }
+
+    /**
+     * Restart the Metrics Emitter to apply new configuration.
+     *
+     * @param event Metric configuration changed event
+     */
+    @Override
+    public void accept(MetricsConfigurationChanged event) {
+        metricsEmitter.restart(event.getConfiguration().isDisableMetrics(),
```

Comment:
related to other comment about stopping, maybe it's clearer to do something like

```
if (disableMetrics) {
  emitter.stop()
} else {
  emitter.restart()
}
```